client/my-sites/checkout/controller.jsx

```diff
@@ -37,6 +39,32 @@ export function checkout( context, next ) {
 	// FIXME: Auto-converted from the Flux setTitle action. Please use <DocumentHead> instead.
 	context.store.dispatch( setTitle( i18n.translate( 'Checkout' ) ) );
 
+	if ( 'right' === abtest( 'showCheckoutCartRight' ) ) {
+		context.store.dispatch( setSection( { name: 'checkout' }, { hasSidebar: false } ) );
+
+		context.primary = (
+			<>
+				<FormattedHeader />
+				<div className="checkout__container">
+					<CheckoutData>
+						<Checkout
+							product={ product }
+							purchaseId={ context.params.purchaseId }
+							selectedFeature={ feature }
+							couponCode={ context.query.code }
+							plan={ plan }
+						/>
+						<CartData>
+							<SecondaryCart selectedSite={ selectedSite } />
+						</CartData>
+					</CheckoutData>
+				</div>
+			</>
+		);
+
+		next();
```

Comment:
We would want to `return` here. Otherwise, the following `context` setup code will be redundantly run even we have set the primary component above.

Reply:
Done in [e1c890d](https://github.com/Automattic/wp-calypso/pull/32832/commits/e1c890d34a64648a263912f153503c137420c8f4).